machine-learning/Dockerfile

```diff
@@ -70,10 +107,18 @@ COPY --from=builder-armnn \
     /opt/ann/build.sh \
     /opt/armnn/
 
+FROM rocm/dev-ubuntu-22.04:6.1.2-complete as prod-rocm
+
+
 FROM prod-${DEVICE} as prod
 
+ARG DEVICE
```

Comment:
I think the environment `DEVICE` was empty before this change. The `DEVICE` only become available until it's declared using `ARG` (see [here](https://docs.docker.com/reference/dockerfile/#scope)

Reply:
You're right, the DEVICE env doesn't get set correctly right now.

Reply:
`main` has this arg set now, so you can remove this.